qa/integration-tests/src/test/java/io/camunda/it/backup/BackupDBClient.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.it.backup;
+
+import io.camunda.search.connect.configuration.DatabaseType;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+/** interface to abstract Elasticsearch/Opensearch clients for testing purposes */
+public interface BackupDBClient extends AutoCloseable {
+  void restore(String repositoryName, Collection<String> snapshots) throws IOException;
+
+  void createRepository(String repositoryName) throws IOException;
+
+  static BackupDBClient create(final String url, final DatabaseType databaseType)
+      throws IOException {
+    return switch (databaseType) {
+      case ELASTICSEARCH -> new ESDBClientBackup(url);
+      case OPENSEARCH -> new OSDBClientBackup(url);
+      default -> throw new IllegalStateException("Unsupported database type: " + databaseType);
+    };
+  }
+
+  void deleteAllIndices(final String indexPrefix) throws IOException;
```

Comment:
💭 We can put a todo to replace cleaning the DB with the purge functionality we have (once it's finished).